src/index.spec.ts

```diff
@@ -1378,3 +1445,20 @@ describe('EppoClient config', () => {
     });
   });
 });
+
+/**
+ * A wrapper for a promise which allows for later resolution.
+ */
+class DeferredPromise<T> {
```

Comment:
❗ woah neato!

Cool concept this would have cleaned up many of the other tests in here had I thought of this 🤦